client/src/pages/HomePage.jsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import ImageGrid from '../components/ImageGrid';
 import ImageUpload from '../components/ImageUpload';
+import TagFilter from '../components/TagFilter';
 
 export default function HomePage() {
+	const [allTags, setAllTags] = useState([]);
+	const [selectedTags, setSelectedTags] = useState([]);
+
+	useEffect(() => {
+		fetch('http://localhost:5000/gettags', {
+			method: 'GET',
+			mode: 'cors',
+		})
+			.then((response) => response.json())
+			.then((data) => {
+				let fetchedTags = [];
+				data.forEach((tag) => {
+					if (!fetchedTags.includes(tag['name'])) {
+						fetchedTags.push({ tag: tag['name'], id: data.indexOf(tag) });
+					}
+				});
+				// sort tags alphabetically
+				fetchedTags.sort((a, b) => (a.tag > b.tag ? 1 : -1));
+				setAllTags(fetchedTags);
+			})
+			.catch((error) => console.log('error', error));
+	}, []);
+
+	useEffect(() => {
+		setSelectedTags(allTags);
```

Comment:
This defaults the drop down to have all tags selected, meaning that if there is a large N number of tags stored, the user would have to uncheck all N-1 other tags to filter by one tag. This also doesn't account for the edge case of images without tags (which wouldn't be shown). 

Maybe initially set no tags selected, and have the default behavior for for when no tags are selected call the API with "*" to get all images.